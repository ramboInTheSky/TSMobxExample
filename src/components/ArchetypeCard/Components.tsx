import styled from '@emotion/styled'
import { Paper, IconButton } from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
    margin-top: ${variables.gutter};
    background-color: ${variables.componentBackgroundClear} !important;
`
export const Header = styled.div`
    background-color: ${variables.archetypeColor};
    color: #333;
    font-size: 1.3rem;
    height: 25px;
    padding: ${variables.indentation};
    display: flex;
    justify-content: space-between;
`

export const Content: React.ComponentType<any> = styled.div`
    background-color: ${variables.componentBackgroundClear} !important;
    padding: ${variables.indentation};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`

export const CloseButton: React.ComponentType<any> = styled(IconButton)`
        padding: 2px !important;
`

export const LockGroup: React.ComponentType<any> = styled(Paper)`
    background-color: ${variables.componentBackground} !important;
    margin: ${variables.indentation};
    padding: ${variables.indentation};
    padding-right: calc(${variables.indentation} + 6px);
`