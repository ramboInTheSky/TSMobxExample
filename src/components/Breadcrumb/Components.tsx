import styled from '@emotion/styled'
import { variables } from '../../constants/theme'

export const Wrapper = styled.div`
    margin-top: ${variables.indentation};
    font-size: 12px;
    /* font-weight: bold; */
	& svg{
        vertical-align: sub;
    }
`
