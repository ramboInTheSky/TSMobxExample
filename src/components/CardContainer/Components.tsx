import styled from '@emotion/styled'
import { Paper, IconButton } from '@material-ui/core'

import { variables } from '../../constants/theme'

export const Wrapper: React.ComponentType<any> = styled(Paper)`
    margin-top: ${variables.gutter};
    background-color: ${variables.componentBackgroundClear} !important;
`
export const Header = styled.div`
    background-color: ${variables.componentBackground};
    color: #333;
    font-size: 1.3rem;
    height: 25px;
    padding: ${variables.indentation};
    display: flex;
    justify-content: space-between;
`

export const Content: React.ComponentType<any> = styled.div`
    background-color: ${variables.componentBackgroundClear} !important;
    padding: ${variables.indentation};
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
`

export const Item: React.ComponentType<any> = styled(Paper)`
    background-color: ${variables.componentBackground} !important;
    margin: ${variables.indentation};
    display: grid;
    grid-template-rows: 32% 50%;
    height: 5rem;
    padding: ${variables.indentation};
    min-width: 10rem;
    max-width: 14rem;
    & div:nth-child(1){
        display: flex;
        flex-direction: row-reverse;
        cursor: pointer;
        font-size: .4rem;
        & svg{
            opacity: .5;
        }
    }
    & div:nth-child(2){
        padding: 5px;
        position: relative;
        font-size: 1.2rem;
        text-align: right; 
        & svg{
            opacity: .2;
            top: -36px;
            left: 0;
            position: absolute;
            font-size: 6rem;
            color: #999;
        }
    }
    
`
export const MenuHeader = styled.div`
    width: 100%;
    background-color: ${variables.dangerColor};
    height: 2rem;
    padding: 1rem;
    color: white;
    font-weight: 700;
    font-size: 1.2rem;
    outline: none;
    & span{
        font-weight: 400;
    }
`

// export const MenuButton: React.ComponentType<any> = styled(IconButton)`

// `

