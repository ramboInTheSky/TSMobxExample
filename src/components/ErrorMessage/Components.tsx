import styled from '@emotion/styled'
import { variables } from '../../constants/theme'

export const Wrapper = styled.div`
	margin: auto;
	border: 5px solid #d43333;
	padding: 2rem;
    background-color: ${variables.componentBackgroundClear};
    margin-bottom: 2.5rem;
    color: #333;
    & h3{
        margin-top:0;
        color: #d43333;
    }
`
