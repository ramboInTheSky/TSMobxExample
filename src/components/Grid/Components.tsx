import styled from '@emotion/styled'
import { variables } from '../../constants/theme'
import Paper from '@material-ui/core/Paper';

export const Wrapper: React.ComponentType<any> = styled(Paper)`
	height: ${variables.columnSize - 12 }vh;
	width: ${variables.columnSize - 2 }vw;
	margin: 2vh auto;
	padding: ${variables.gutter};
	padding-bottom: calc(${variables.gutter} * 2);
`

export const SmallWrapper: React.ComponentType<any> = styled(Paper)`
	height: ${variables.columnSize - 40 }vh;
	width: ${variables.columnSize - 2 }vw;
	margin: 2vh auto;
	padding: ${variables.gutter};
	padding-bottom: calc(${variables.gutter} * 3);
`