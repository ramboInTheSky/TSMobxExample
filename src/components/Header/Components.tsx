import styled from '@emotion/styled'
import IconButton from '@material-ui/core/IconButton'

import {variables} from '../../constants/theme'

export const Wrapper = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	background-color: ${variables.componentBackground};
	padding: 0.3rem;
`
export const Logo = styled.img`
	width: 150px;
	height: 50px;
`

export const Title = styled.div`
	font-size: 2rem;
	font-weight: bold;
	margin: auto;
`

export const MenuButton: React.ComponentType<any> = styled(IconButton)`
	margin-left: 110px !important;
`
