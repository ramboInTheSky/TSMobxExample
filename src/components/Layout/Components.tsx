import styled from '@emotion/styled'
import Paper from '@material-ui/core/Paper'
import { variables } from '../../constants/theme'

export const Wrapper = styled.div`
	display: grid;
	width: 100%;
	/* text-align: center; */
	justify-items: center;
	grid-row-gap: 2rem;
`

export const Heading = styled.div`
	display: flex;
	width: ${variables.columnSize}vw;
	flex-direction: column;
	justify-content: space-around;
`

export const Content = styled.div`
	display: flex;
	width: ${variables.columnSize}vw;
	flex-direction: column;
	justify-content: space-around;
	/* background-color: ${variables.componentBackground}; */
`