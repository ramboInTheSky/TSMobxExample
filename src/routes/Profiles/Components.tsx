import styled from '@emotion/styled'
import IconButton from '@material-ui/core/IconButton'
import { variables } from '../../constants/theme'

export const ActionBar = styled.div`
	display: flex;
	flex-direction: row;
	padding-top: ${variables.gutter};
	justify-content: space-between;
`
export const SearchWrapper = styled.div`
	width: 84%;
	padding-left: ${variables.gutter};
`
export const Action: React.ComponentType<any> = styled.div`
	margin: auto !important;
	padding-top: 2px !important;
`

export const ActionIcon: React.ComponentType<any> = styled(IconButton)`
	margin: auto !important;
	padding-top: 2px !important;
`

export const Title = styled.div`
	padding-left: ${variables.gutter};
	font-size: ${variables.headerFontSize};
	font-weight: bold;
	margin-top: ${variables.gutter};
`
