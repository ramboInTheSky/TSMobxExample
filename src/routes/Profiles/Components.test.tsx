import React from 'react'
import { shallow } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import { ActionBar, SearchWrapper, Action, Title, ActionIcon } from './Components'

describe('<ActionBar />', () => {
	it('renders correctly', () => {
		expect(shallow(<ActionBar />)).toMatchSnapshot()
	})
})

describe('<SearchWrapper />', () => {
	it('renders correctly', () => {
		expect(shallow(<SearchWrapper />)).toMatchSnapshot()
	})
})

describe('<Action />', () => {
	it('renders correctly', () => {
		expect(shallow(<Action />)).toMatchSnapshot()
	})
})

describe('<Title />', () => {
	it('renders correctly', () => {
		expect(shallow(<Title />)).toMatchSnapshot()
	})
})

describe('<ActionIcon />', () => {
	it('renders correctly', () => {
		expect(shallow(<ActionIcon />)).toMatchSnapshot()
	})
})

