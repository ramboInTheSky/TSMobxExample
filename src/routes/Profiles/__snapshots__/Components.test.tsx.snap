// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Action /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<ActionBar /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<ActionIcon /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<SearchWrapper /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<Title /> renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
